File: Backend/IPC2Proyecto2/backendRevistas/src/main/java/Modelo/Setting.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Modelo;

/**
 *
 * @author ordson
 */
public class Setting {
    private int codigo;
    private double porcentajeCobro;
    private double cuotaDiaria;
    private double precioHoraAnuncio;

    public Setting(int codigo, double porcentajeCobro, double cuotaDiaria, double precioHoraAnuncio) {
        this.codigo = codigo;
        this.porcentajeCobro = porcentajeCobro;
        this.cuotaDiaria = cuotaDiaria;
        this.precioHoraAnuncio = precioHoraAnuncio;
    }

    public Setting(double porcentajeCobro, double cuotaDiaria, double precioHoraAnuncio) {
        this.porcentajeCobro = porcentajeCobro;
        this.cuotaDiaria = cuotaDiaria;
        this.precioHoraAnuncio = precioHoraAnuncio;
    }

    public Setting() {
    }

    public double getPrecioHoraAnuncio() {
        return precioHoraAnuncio;
    }

    public void setPrecioHoraAnuncio(double precioHoraAnuncio) {
        this.precioHoraAnuncio = precioHoraAnuncio;
    }

    public int getCodigo() {
        return codigo;
    }

    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }

    public double getPorcentajeCobro() {
        return porcentajeCobro;
    }

    public void setPorcentajeCobro(double porcentajeCobro) {
        this.porcentajeCobro = porcentajeCobro;
    }

    public double getCuotaDiaria() {
        return cuotaDiaria;
    }

    public void setCuotaDiaria(double cuotaDiaria) {
        this.cuotaDiaria = cuotaDiaria;
    }

    @Override
    public String toString() {
        return "Setting{" + "codigo=" + codigo + ", porcentajeCobro=" + porcentajeCobro + ", cuotaDiaria=" + cuotaDiaria + ", precioHoraAnuncio=" + precioHoraAnuncio + '}';
    }
    
}

File: Frontend/IPC2Proyecto2/src/app/objetos/Setting.ts
export class Setting{
    _codigo!:number;
    _porcentajeCobro!:number;
    _cuotaDiaria!:number;
    _precioHoraAnuncio!:number;
    constructor(){}
    
    get codigo(){
        return this._codigo;
    }
    set codigo(codigo:number){
        this._codigo = codigo;
    }
    get porcentajeCobro(){
        return this._porcentajeCobro;
    }
    set porcentajeCobro(porcentajeCobro:number){
        this._porcentajeCobro = porcentajeCobro;
    }
    get cuotaDiaria(){
        return this._cuotaDiaria;
    }
    set cuotaDiaria(cuotaDiaria:number){
        this._cuotaDiaria= cuotaDiaria;
    }
    get precioHoraAnuncio(){
        return this._precioHoraAnuncio;
    }
    set precioHoraAnuncio(precioHoraAnuncio:number){
        this._precioHoraAnuncio = precioHoraAnuncio;
    }
}